error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/shadowing.rs:6:11
   |
6  | #[ensures(result != 3)] 
   |           ^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/shadowing.rs:7:1
   |
7  | / fn foo(a: i32, b: i32) -> i32 {
8  | |     if a == 42 {
9  | |         let b = 3;
10 | |         b
...  |
14 | |     }
15 | | }
   | |_^
note: counterexample for "a"
       value:   42
  --> $DIR/shadowing.rs:7:8
   |
7  | fn foo(a: i32, b: i32) -> i32 {
   |        ^
note: counterexample for "b"
       value:   0
  --> $DIR/shadowing.rs:7:16
   |
7  | fn foo(a: i32, b: i32) -> i32 {
   |                ^
note: counterexample for "b"
       value:   3
  --> $DIR/shadowing.rs:9:17
   |
9  |         let b = 3;
   |                 ^
note: counterexample for "result"
       value:   3
  --> $DIR/shadowing.rs:10:9
   |
10 |         b
   |         ^

error: aborting due to previous error

